### PR TITLE
fixed type error and added harmonic intervals

### PR DESCRIPTION
**Pull request by yann-lecostaouec**
I fixed two small typescript errors and added a setting to the interval exercise to allow harmonic or melodic intervals.

### CODE REVIEW

File: src/app/exercise/exercises/IntervalExercise/intervalExercise.ts [hunk 1]
```diff
@@ -133,11 +135,17 @@ export const intervalExercise = () => {
       const randomStartingNoteNumber: NoteNumber = _.random(range.lowestNoteNumber, range.highestNoteNumber - randomIntervalDescriptor.semitones);
       let lowNoteName = toNoteName(randomStartingNoteNumber);
       let highNoteName = toNoteName(randomStartingNoteNumber + randomIntervalDescriptor.semitones);
-      let [startNoteName, endNoteName] = _.shuffle([lowNoteName, highNoteName])
+      let [startNoteName, endNoteName] = _.shuffle([lowNoteName, highNoteName]);
+      let toPlay;
+      if (settings.intervalType === 'melodic')
+        toPlay = [{ notes: startNoteName }, { notes: endNoteName } ]
+      else
+        toPlay = [{ notes: [startNoteName, endNoteName] }]
```
Comment (ShacharHarshuv) [inline src/app/exercise/exercises/IntervalExercise/intervalExercise.ts:143]:
```suggestion
      const partToPlay = settings.intervalType === 'melodic' ? 
        [{ notes: startNoteName }, { notes: endNoteName } ] :
        [{ notes: [startNoteName, endNoteName] }];
```

A little bit more elegant.

**Comment by ShacharHarshuv** [comment]
Hey @yann-lecostaouec, thanks so much for contributing! I've added a few comments.

**Comment by yann-lecostaouec** [comment]
@ShacharHarshuv I checked in updates to address your comments. Thanks for making open-ear and open-sourcing it!

**Review by ShacharHarshuv — APPROVED** [review]
Looks good! Thank you for your contribution! I'll build a new version as soon as possible. (But you can already build one locally and deploy it to your phone as a test build)

**Pull request merged**